### PR TITLE
added anagram algorithm in java

### PR DESCRIPTION
**Pull request by rajeshpanjiyar**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Yeah I have.)

### Description

(This PR is to add an anagram algorithm in java of different approaches.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(This is to add an anagram algorithm in java under the issue#3709.)

### CODE REVIEW

File: Java/Algorithms/Anagram.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		
+      Scanner scan = new Scanner(System.in);
+		
+	System.out.println("Enter a string");
+	 String string1 = scan.nextLine();        //takes a input string
+		
+	    System.out.println("Enter another string");
+	      String string2 =scan.nextLine();     //takes another input string
+				scan.close();
+		   boolean isAnagram =true;
```
Comment (ritvij14) [inline Java/Algorithms/Anagram.java:17]: The java code has too many indentations. Please format the code properly.

File: Java/Algorithms/Anagram.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	 }else {
+	System.out.println("not anagram");
+			   }
+			
+		
+		}
+	  }
+
+		
+
```
Comment (ritvij14) [inline Java/Algorithms/Anagram.java:48]: Remove extra lines, there is only one line needed at EOF (End of File). Also, add comments to show input and output, and space and time complexity.

**Review by ritvij14 — CHANGES_REQUESTED** [review]

**Comment by rajeshpanjiyar** [comment]
Yeah sure

**Comment by HarshCasper** [comment]
Closing the PR because of no updates.

**Comment by rajeshpanjiyar** [comment]
Hey @HarshCasper, I had regenerated pr for this. Can you please review this.
https://github.com/TesseractCoding/NeoAlgo/pull/4002

**Pull request closed without merging**